PerimeterXModule/Internals/PxContext.cs

```diff
@@ -190,30 +190,36 @@ public PxContext(HttpContext context, PxModuleConfigurationSection pxConfigurati
 			CustomBlockUrl = pxConfiguration.CustomBlockUrl;
 			RedirectOnCustomUrl = pxConfiguration.RedirectOnCustomUrl;
 
-            MonitorRequest = shouldMonitorRequest(context.Request.Url.AbsolutePath, pxConfiguration);
+            MonitorRequest = shouldMonitorRequest(context, pxConfiguration);
 		}
 
-        private bool shouldMonitorRequest(String uri, PxModuleConfigurationSection pxConfiguration)
+        private bool shouldMonitorRequest(HttpContext context, PxModuleConfigurationSection pxConfiguration)
         {
-            if (uri.IndexOf("/", StringComparison.Ordinal) == 0) 
+            string uri = context.Request.Url.AbsolutePath;
+            if (uri.IndexOf("/", StringComparison.Ordinal) == 0)
             {
                 uri = uri.Substring(1);
             }
 
-			var mitigationUrls = pxConfiguration.MitigationUrls;
-			if (mitigationUrls.Count > 0)
-			{
-				if (mitigationUrls.Contains(uri))
-				{
-					return false;
-				}
-				return true;
-			}
-			else
-			{
-				return pxConfiguration.MonitorMode;
-			} 
-           
+            var mitigationUrls = pxConfiguration.MitigationUrls;
+            if (mitigationUrls.Count > 0)
+            {
+                if (mitigationUrls.Contains(uri))
```

Comment:
I know it's not your code originally, but lines 207 to 211 can be replaced with `return !mitigationUrls.Contains(uri)`

Reply:
I thought you referred to this party also: `if (mitigationUrls.Count > 0)`. Will fix.